### PR TITLE
Breaking functions to be more modular

### PR DESCRIPTION
**Pull request by row49382**
**Branch name doesn't represent the work done**

- Implemented Listing of Enum values before entering a value so that the user knows which enum values are valid
- Implemented base work for the add proficiencies work.
- Broke up functions in the commands to be more modular.

### CODE REVIEW

File: DungeonAndDragonsRollStats/src/main/java/enums/CharacterGender.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package enums;
+
+public enum CharacterGender {
+	Male,
+	Female,
+	Non_Binary
```
Comment (binkky99) [inline DungeonAndDragonsRollStats/src/main/java/enums/CharacterGender.java:6]: lols, way to be progressive

File: DungeonAndDragonsRollStats/src/main/java/enums/CharacterRace.java [hunk 2]
```diff
@@ -84,12 +87,45 @@ public String toString() {
...
+
+    /**
+     * Setups the ability score increases that will be applied
+     * based on proficiency/expertise
+     * 
+     * Still need to design means for implementation
+     * @param abilityScoreIncreases
+     * @return 
+     */
+	private static Map<Ability, Integer> setupAbilityScoreIncreases(List<AbstractMap.SimpleEntry<Ability, Integer>> abilityScoreIncreases) {
```
Comment (binkky99) [inline DungeonAndDragonsRollStats/src/main/java/enums/CharacterRace.java:112]: is this just initializing a map for the ability scores?  Should it just be called setupAbilityScores?  I could be misunderstanding what this is doing.

File: DungeonAndDragonsRollStats/src/main/java/models/Player.java [hunk 3]
```diff
@@ -151,8 +159,22 @@ public PlayerBuilder setAbilityStats(Map<String, Integer> abilityStats) {
...
 		}
 		
+		public PlayerBuilder setProficiencies(Map<String, Integer> proficiencies) {
+			this.proficiencies = proficiencies;
+			return this;
+		}
+		
 		public Player build() {
-			return new Player(characterClass, characterRace, characterSubClass, characterSubRace, playerName, characterName, startingLevel, abilityStats);
+			return new Player(
```
Comment (binkky99) [inline DungeonAndDragonsRollStats/src/main/java/models/Player.java:168]: I like how this is organized.

**Review by binkky99 — COMMENTED** [review]
I can't see any major red flags. I noted the proficiency logic thing, we probably just want to have basic functionality for adding proficiencies and go from there.

**Comment by binkky99** [comment]
done?

**Comment by binkky99** [comment]
i don't know how to accept this Bobby

**Review by binkky99 — APPROVED** [review]

**Comment by binkky99** [comment]
I did it!

File: DungeonAndDragonsRollStats/src/main/java/enums/CharacterRace.java (hunk 2 above)
Comment (row49382) [reply DungeonAndDragonsRollStats/src/main/java/enums/CharacterRace.java:112]: This is for giving each enum value the appropriate ability score increase based on the class/race. For example, If I'm an Elf, my Dexterity is +2, so then the map they would maintain would look something like this:

```java
Map<Ability, Integer> abilityScoreIncreases = new HashMap<Ability, Integer>();
abilityScoreIncreases.put(Ability.Dexterity, 2);
```
so then when you calculate the character's ability score, it uses the races/classes values to sum up the final result.

**Pull request merged**